packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -153,7 +158,7 @@ class DateTimePicker extends React.Component {
     super(props);
 
     this.state = {
-      dateTime: DateTimeUtils.createSafeDate(props.value),
+      dateTime: DateTimeUtils.createSafeDate(props.value, props.timeZone),
```

Comment:
As discussed offline, the timezone will need to be stored in the state since consumers could set a different timezone after the initial mount. `this.state.dateTime` will be updated to the new timezone should it change.

Reply:
Updated here 1e47747132f83358d17d2aaed1f98c6bd4a5cfaf